CONTRIBUTING.md

```diff
@@ -62,6 +62,26 @@ Once it succeeds, one has to go back to (view 1) and re-run the failed jobs, so
 
 After doing these steps once, for other PRs, one only needs to re-run deep checks in (view 1)
 
+### How to fix nightly build failures / check deep tests?
+
+- Create a branch (e.g. `fix-ci-<date>`) that should fix the nightly build.
+  (Either an actual fix, or reverting a PR that caused the problem - look at the logs)
+- Push your branch to the main Dafny repository.
+- After pushing, you'll get a link to create a PR, click on it.
+- **Before clicking on the "Create PR" button**, add the label `run-deep-tests`. If you add it later, you'll have to push a new commit to trigger the deep tests.
+- Click on the "Create PR" button.
+  The CI will run the deep tests within this PR (and for every new commit pushed to the branch). Note that only the "required" tests count.
+- If some required tests fail, investigate and push more commits, but know that each CI run takes a lot of time, so try to fix the problem with as few commits as possible (ideally 1)
+  It is also possible you find some tests that fail randomly, in that case, feel free to relaunch them again and report to the team.
```

Comment:
```suggestion
  It is also possible you find some tests that fail randomly, in that case, re-run them and report to the team. Avoid using the re-run all failed jobs here, because it will re-run _all_ jobs, so select them individually instead.
```